File: APISolidarityManager/APISolidarityManager/Entities/Item.cs
namespace APISolidarityManager.Models
{
    public class Item
    {
        public Guid Id { get; set; }
        public Guid CategoryId { get; set; }
        public Guid ItemTemplateId { get; set; }
        public string Name { get; set; } = null!;
        public string? Brand { get; set; }
        public decimal PackageQuantity { get; set; }
        public decimal TemplateWeight { get; set; }
        public string? UnitOfMeasure { get; set; }
        public string? Notes { get; set; }
        public bool Active { get; set; } = true;
        public DateTime CreatedAt { get; set; }
        public DateTime? UpdatedAt { get; set; }
        public ItemCategory Category { get; set; } = null!;
        public ItemTemplate ItemTemplate { get; set; } = null!;

        public ICollection<InventoryBatch> InventoryBatches { get; set; } = new List<InventoryBatch>();
    }
}

File: donation-system-web/src/types/item.ts
export interface ItemResponse {
  id: string;
  categoryId: string;
  categoryName: string;
  itemTemplateId: string;
  itemTemplateName: string;
  name: string;
  brand?: string;
  packageQuantity: number;
  templateWeight: number;
  unitOfMeasure?: string;
  notes?: string;
  active: boolean;
  createdAt: string;
  updatedAt?: string;
}

export interface ActiveItemResponse {
  id: string;
  name: string;
  brand?: string;
  packageQuantity: number;
  templateWeight: number;
  unitOfMeasure?: string;
  categoryId: string;
  categoryName: string;
  itemTemplateId: string;
  itemTemplateName: string;
  needGroup: string;
  referenceQuantity: number;
  totalQuantity: number;
}

export interface UpdateItemRequest {
  categoryId: string;
  itemTemplateId: string;
  name: string;
  brand?: string;
  packageQuantity: number;
  templateWeight: number;
  unitOfMeasure?: string;
  notes?: string;
  active: boolean;
}

export interface CreateItemRequest {
  categoryId: string;
  itemTemplateId: string;
  name: string;
  brand?: string;
  packageQuantity: number;
  templateWeight: number;
  unitOfMeasure?: string;
  notes?: string;
  active: boolean;
}
